### PR TITLE
[8.1] Remove VMDIRAC

### LINKED ISSUE #5282: CloudDirector -> SiteDirector
CloudDirector was first designed similar to SiteDirector, but the latter evolved in the meantime. 
We should check:

1. if we could avoid having at all this CloudDirector
2. if CluodDirector will keep being "special", then we can see how much the current implementation can inherit from SiteDirector

### PR DESCRIPTION
**Pull request by sfayer**
Here is the patch to remove VMDIRAC. I'm creating it as a draft initially until it's announced/discussed at a BiLD meeting.

Closes #5282.

Regards,
Simon

BEGINRELEASENOTES
*WorkloadManagement
CHANGE: Remove VMDIRAC components
ENDRELEASENOTES

### CODE REVIEW

**Comment by fstagni** [comment]
Naively I thought that the VirtualMachineDB and VirtualMachineManager service (and related commands) would still be needed, but clearly this is not the case (of course the CloudComputingElement knows nothing about them). Not having any operational experience with them I can not judge, but were them somehow useful?
This BTW also means that some WebAbbDIRAC bits can be removed too.

**Comment by sfayer** [comment]
The VirtualMachineDB was mostly used for tracking the state of instances, the same basic information is now stored/mapped in the Pilot DB and uses the associated commands. For example instead of using the dirac-vm-instance-stop command, you'd now use dirac-admin-kill-pilot with the VM's pilot reference (although there is rarely a good operational reason to do that). dirac-vm-get-pilot-output (which used the IP field(s) in the database and was unreliable in some instances) will be replaced by the generic remote pilot logging that Janusz is working on.

A lot of the state in VirtualMachineDB was focused around managing and tuning the number of running instances in a group of VMs that it called a "pod". The pod code has been broken for years (and was mostly removed in the merge) so no-one can be using it like that any more. The modern usage is dynamic where new instances are started as required and stopped when no-longer running a JobAgent, this means the manual VM management commands are pretty much obsolete.

If it turns out there are fields that people care about, I'm sure we can discuss some way to store them even if it means putting a greatly simplified version of the DB back: We can't be 100% sure until all of the communities have at least tested CloudCE against their requirements but we have attempted to accommodate every use case we know about.

File: src/DIRAC/ConfigurationSystem/Client/Helpers/Resources.py [hunk 1]
```diff
@@ -446,152 +446,6 @@ def getInfoAboutProviders(of=None, providerName=None, option="", section=""):
...
-    if tags:
-        resultDict["Tag"] = tags
-    if reqtags:
-        resultDict["RequiredTag"] = reqtags
-    resultDict["VMType"] = vmtype
-    resultDict["Site"] = site
-    return S_OK(resultDict)
-
-
 def getPilotBootstrapParameters(vo="", runningPod=""):
```
Comment (fstagni) [inline src/DIRAC/ConfigurationSystem/Client/Helpers/Resources.py]: This method can also be removed.

**Comment by fstagni** [comment]
You need to remove these lines:

https://github.com/DIRACGrid/DIRAC/blob/integration/tests/Integration/all_integration_server_tests.sh#L53
https://github.com/DIRACGrid/DIRAC/blob/integration/tests/Integration/all_integration_client_tests.sh#L61

**Review by fstagni — APPROVED** [review]

**Comment by fstagni** [comment]
Looks all good and I can merge. There are few updates to do, will you do or should I?

- [x] https://trello.com/c/Y7qSO94Z
- [x] update https://github.com/DIRACGrid/DIRAC/wiki/DIRAC-8.1
- [x] CloudDirector + DB can be uninstalled from lbdiraccertif machine according to what will in the bullet above

**Comment by marianne013** [comment]
@fstagni I've removed VMDIRAC from lbcertifdirac70.cern.ch, and updated Trello and the Wiki.

**Pull request merged**